### ISSUE TITLE
track project collaborators and managers by dn, not username

### ISSUE THREAD

**Issue by anderbubble** [ResearchComputing/RCAMP#138]
I noticed that, in rcamp2, there were two sampedro accounts: one in ucb, and one in csu. This is precisely the kind of username conflict that the separate ous is designed to support and differentiate; but the project collaborator and manager relationship is currently keyed on the username, which is ambiguous. This field should store a list of DNs, not a list of usernames.

**Comment by anderbubble** [comment]
I fear this might end up being a bigger issue as we start using RCAMP to manage group membership for CSU users. Right now, doesn't it add users to groups by UID? But the CSU username isn't actually their UID, it's UID@colostate.edu.

We need to ensure that adding CSU users to both CSU and CU groups produces a membership based on UID@colostate.edu

**Comment by zebulasampedro** [comment]
@anderbubble If we're currently distinguishing CSU users on {UID}@{ORG_SUFFIX}, will this issue present any problems in practice? Also since we want to start moving the codebase back to a flat LDAP hierarchy this will just end up being more logic to migrate.

Also, do we want to change the way that users are presented to project managers when they are filling out the forms?

This issue is absolutely tractable, I just want to make sure it is worth doing in the long run.

**Comment by anderbubble** [comment]
tl;dr, maybe we should just make sure that @pruprecht and @frahm are aware of this issue and fix it with the previously-discussed reunification of UCB and CSU OUs.

@zebulasampedro the problem is that while CSU users are distinguished on the Linux side as {UID}@{ORG_SUFFIX}, RCAMP doesn't. So when it generates a user list for adding to a project or group, you only see the flat uid space, with potential conflicts.

If you add a csu user to a csu group, I think sssd is appending @colostate.edu to both the membership list and the group name (that's what I'm seeing, anyway); but that means you can't add a UCB user to a CSU group (because it would become `$ucb_uid@colostate.edu` in `sssd`, nothing you can do about that) or a CSU user to a UCB group (which would necessitate adding the `@colostate.edu` suffix manually).

So maybe we should ignore the LDAP part of this issue, and just accept that, for now, we can't cross domain boundaries with group membership. (That should be fixed sooner than later, but maybe not *immediately*.)

But that doesn't change the fact that the portal itself doesn't distinguish between the same uid coming from different ldap objects when tracking project membership. This would get sidestepped when we reunify ldap OUs; but it's still the right thing to make sure that any internal RCAMP reference to an LDAP object is referenced by DN, not by a single attribute, even uid. (It should still *display* something more friendly, like UID, but the UI should be able to distinguish between them and use the DN on the backend.)
